### ISSUE TITLE
Add support for Git with Kerberos authentication for SCM project.

### ISSUE THREAD

**Issue by IshwarKanse** [ansible/awx#5925]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
We have a use case from a customer where they want to use Git with Kerberos authentication. We tested the following workaround which is working fine.

* Manually create a  ticket for the Kerberos user from the awx user on all the Tower nodes.
* Add a project on Tower with the SCM URL.without attaching any cred.
* Disable job isolation and run the project sync.
* The job will use the Kerberos ticket that was manually created for project sync.

The issue here is that we need to disable isolation, ticket needs to be created manually on all the nodes. Also we cannot specify a user from the SCM project for which a ticket will be created on the fly.

**Comment by wenottingham** [comment]
Just to be clear: this is GSSAPI git+ssh, or a different connection method?

**Comment by IshwarKanse** [comment]
Yeah, its GSSAPI git+https.

**Comment by ryanpetrello** [comment]
@wenottingham I'll let you decide where you want to place this.

**Comment by wenottingham** [comment]
GSSAPI *https*? @IshwarKanse , Are you sure it's not cloning over ssh?

**Comment by IshwarKanse** [comment]
@wenottingham yeah, they are not using SSH. I've sent you a message on Slack with the details. The customer is also monitoring the RFE, if we want to get more details about their Kerberos implementation, we can ask him or I can get the details on the case.
